### ISSUE TITLE
Allow `dataPath` to be provided for `Web5.connect`

### ISSUE THREAD

**Issue by sondreb** [TBD54566975/web5-js#949]
This is a request to add `dataPath` as a parameter for Web5.connect() method. The dataPath should be forwarded to this request within the `connect` method:

```const userAgent = await Web5UserAgent.create({ agentVault });```

If this `create` method is provided with `dataPath`, it could be possible to have separate databases locally on the same app.

**Comment by bnonni** [comment]
Are you asking to bubble up `dataPath` as a direct arg in the `Web5.connect()` method call? If so, this seems redundant because `await Web5UserAgent.create()` takes a `dataPath` arg. See https://github.com/TBD54566975/web5-js/blob/bfa0417a2e9fc1300c5e604bea19b75ab1c73645/packages/user-agent/src/user-agent.ts#L151-155

You can achieve the resule you want without adding `dataPath` to `.connect` by using this method.

### Examples
1. Multiple agents in one `DATA` folder
```ts
const agent0 = await Web5UserAgent.create({ dataPath: 'DATA/AGENT_0' });
const agent1 = await Web5UserAgent.create({ dataPath: 'DATA/AGENT_1' });
```

2. Multiple `DATA` folders each with its own agent 
```ts
const agent0 = await Web5UserAgent.create({ dataPath: 'DATA_0/AGENT' });
const agent1 = await Web5UserAgent.create({ dataPath: 'DATA_1/AGENT' });
```

**Comment by sondreb** [comment]
> Are you asking to bubble up `dataPath` as a direct arg in the `Web5.connect()` method call?

Yes, just make it easier for consumers of the library to supple it, without having to replicate some of the logic within the `connect` method. Thanks for the examples given👍

**Comment by bnonni** [comment]
> > Are you asking to bubble up `dataPath` as a direct arg in the `Web5.connect()` method call?
> 
> Yes, just make it easier for consumers of the library to supple it, without having to replicate some of the logic within the `connect` method. Thanks for the examples given👍

@sondreb I understand. Makes sense. Obv I can't speak for the web5-js dev team, but speaking for myself, if I were them, I'd want to avoid overloading `.connect` method. It already has a sprawling list of args and does a LOT.

https://github.com/TBD54566975/web5-js/blob/bd1cb00616029b0d18687b597e90d3b7c4dbeae1/packages/api/src/web5.ts#L253-L264

But maybe adding it to one of the current args objects would solve your problem and keep the interface intact.

For example, the `didCreateOptions` object
```ts
export type DidCreateOptions = {
  /** Override default dwnEndpoints provided during DID creation. */
  dwnEndpoints?: string[];
 agentDataPath?: string; // Pass a custom dataPath down to Web5UserAgent.create and new HdIdentityVault
}
```

This small change would keep the `.connect` method arg interface consistent across versions while adding this optional var to pass into `didCreateOptions` to achieve your desired outcome.

Example implementation
```ts
const connection0 = await Web5.connect({
    password: 'password0',
    didCreateOptions: { agentDataPath: 'USER0_DATA/AGENT' } // or whatever you want
});

const connection1 = await Web5.connect({
    password: 'password1',
    didCreateOptions: { agentDataPath: 'DATA/USER_AGENT_1' }
});
```

From personal experience, this will move faster if you take the initiative to write the code and submit a PR :) Happy to collab with you on it in Discord if you want. Just lmk.

**Comment by sondreb** [comment]
Thanks for the reply @bnonni, though the connect function is already implemented using a structured parameter and doesn't require additional connect overloads. JavaScript doesn't support function overloading; your point would be valid if it was using basic arguments.

It's not a good alternative to put it on `didCreateOptions`, as this is not related to the DID; but the DWN.

Should be fairly easy to make a PR for this, you could give it a go if you want to, and I'll review and help?

**Comment by bnonni** [comment]
@sondreb Sorry, I misused the term overload. I simply meant, that method has a lot of args already, and that can make the interface messy and confusing leading to bad dev experience, unexpected user behavior and/or unexpected use cases resulting in frustration. More of an intuition than anything else, but regardless, I'll let you take lead on it. Feel free to ping me if you'd like review and help.

**Comment by shamilovtim** [comment]
@sondreb If we let you pass your own userAgent into the method would that solve this for you? I think it would be the most elegant solution here. Then you can add whatever optionality / params to your UserAgent that you want. It would also be healthier for the codebase since it would eliminate the `agentVault` param.

**Comment by sondreb** [comment]
Maybe it would be enough, at least making it a little bit easier than it is now. Thanks!

**Comment by shamilovtim** [comment]
@sondreb open for PRs until someone else gets to it if it's a priority for you. otherwise one of myself or @LiranCohen  could eventually solve.

**Comment by bnonni** [comment]
> [@sondreb](https://github.com/sondreb) open for PRs until someone else gets to it if it's a priority for you. otherwise one of myself or [@LiranCohen](https://github.com/LiranCohen) could eventually solve.

I'll give this a shot today.
